### ISSUE TITLE
"Undefined" Group label when 2 months ago is in previous year

### ISSUE THREAD

**Issue by benoitgrelard** [leggett/simplify#452]
Hi @leggett, I started trying out the new version on Safari today and so far it's really great.

One thing I have noticed is I get an "undefined" label in my particular case, not sure what it should have been or if they should have been grouped in the "Earlier" group.

Happy to provide more info about actual dates on each emails as you need.

<img width="122" alt="image" src="https://user-images.githubusercontent.com/1539897/103558236-43934b00-4eac-11eb-828c-315fdd18e42b.png">

**Comment by leggett** [comment]
Interesting!! Can you attach a screenshot of the date column? Or you can email me if you don't want to post here at hi@simpl.fyi

**Comment by benoitgrelard** [comment]
<img width="1180" alt="Screenshot 2021-01-04 at 17 00 20" src="https://user-images.githubusercontent.com/1539897/103559689-72aabc00-4eae-11eb-87f3-c2b671a3e5da.png">

**Comment by leggett** [comment]
Weird weird weird. Are you at all familiar with the developer console? If I
gave you some commands to run, would you be willing to tell me what the
output is?

**Comment by benoitgrelard** [comment]
I'm a web developer myself so I can do whatever you need me to 😉

**Comment by leggett** [comment]
Awesome. Can you run this from your inbox:

```
document.querySelectorAll("div[role='main'] .ae4:not([style*='none']) .Cp:not(.adverts) tr.zA").forEach((thread, i) => {
  let date = thread.querySelector(".xW > span")?.title;
  let dateAttr = thread.getAttribute("date");
  console.log(i, date, dateAttr);
});
```

**Comment by benoitgrelard** [comment]
Here we go:

```
[Log] 0 – "Mon, Jan 4, 2021, 1:23 PM" – "today"
[Log] 1 – "Fri, Jan 1, 2021, 9:55 PM" – "month0"
[Log] 2 – "Fri, Jan 1, 2021, 12:04 PM" – "month0"
[Log] 3 – "Thu, Dec 17, 2020, 10:46 AM" – "month1"
[Log] 4 – "Tue, Dec 1, 2020, 3:33 PM" – "month1"
[Log] 5 – "Sat, Nov 14, 2020, 1:54 PM" – "month2"
[Log] 6 – "Tue, Nov 10, 2020, 5:39 PM" – "month2"
[Log] 7 – "Tue, Nov 10, 2020, 3:18 PM" – "month2"
[Log] 8 – "Fri, May 29, 2020, 9:48 AM" – "earlier"
```

**Comment by leggett** [comment]
Duh, I see the problem. It should say November but I'm doing something wrong with the array of months. Will fix. Thank you!!

**Comment by leggett** [comment]
I have a fix checked in for 2.1.17. Will go out this week. Thanks for the report and help.
